src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/BodySubstitutionParser.cs

```diff
@@ -1,174 +1,176 @@
+                : base(context, documentStream, resource, resourceAssembly, xmlDocumentLocation, featureSwitchValues)
         {
+            _methodSubstitutions = new Dictionary<MethodDesc, BodySubstitution>();
+            _fieldSubstitutions = new Dictionary<FieldDesc, object>();
         }
 
-        public void Parse(SubstitutionInfo xmlInfo)
-        {
-            _substitutionInfo = xmlInfo;
-            bool stripSubstitutions = _context.IsOptimizationEnabled(CodeOptimizations.RemoveSubstitutions, _resource?.Assembly);
-            ProcessXml(stripSubstitutions, _context.IgnoreSubstitutions);
-        }
-
-        protected override void ProcessAssembly(AssemblyDefinition assembly, XPathNavigator nav, bool warnOnUnresolvedTypes)
+        protected override void ProcessAssembly(ModuleDesc assembly, XPathNavigator nav, bool warnOnUnresolvedTypes)
         {
             ProcessTypes(assembly, nav, warnOnUnresolvedTypes);
-            ProcessResources(assembly, nav);
         }
 
-        protected override TypeDefinition? ProcessExportedType(ExportedType exported, AssemblyDefinition assembly, XPathNavigator nav) => null;
+        // protected override TypeDesc? ProcessExportedType(ExportedType exported, ModuleDesc assembly, XPathNavigator nav) => null;
 
-        protected override bool ProcessTypePattern(string fullname, AssemblyDefinition assembly, XPathNavigator nav) => false;
+        protected override bool ProcessTypePattern(string fullname, ModuleDesc assembly, XPathNavigator nav) => false;
 
-        protected override void ProcessType(TypeDefinition type, XPathNavigator nav)
+        protected override void ProcessType(TypeDesc type, XPathNavigator nav)
         {
             Debug.Assert(ShouldProcessElement(nav));
             ProcessTypeChildren(type, nav);
         }
 
-        protected override void ProcessMethod(TypeDefinition type, XPathNavigator methodNav, object? _customData)
+        protected override void ProcessMethod(TypeDesc type, XPathNavigator methodNav, object customData)
         {
-            Debug.Assert(_substitutionInfo != null);
             string signature = GetSignature(methodNav);
             if (string.IsNullOrEmpty(signature))
                 return;
 
-            MethodDefinition? method = FindMethod(type, signature);
+            MethodDesc method = FindMethod(type, signature);
             if (method == null)
             {
-                LogWarning(methodNav, DiagnosticId.XmlCouldNotFindMethodOnType, signature, type.GetDisplayName());
+                // LogWarning(methodNav, DiagnosticId.XmlCouldNotFindMethodOnType, signature, type.GetDisplayName());
                 return;
             }
 
             string action = GetAttribute(methodNav, "body");
-            switch (action)
+            if (!String.IsNullOrEmpty(action))
```

Comment:
There's an addition of `if (!String.IsNullOrEmpty(action))` that doesn't exist in ILLink.

Seems like if there's no `body` attribute, ILLink would go to the `default` case and warn whereas this would silently ignore such entry. We should keep ILLink structure.